core/templates/dev/head/pages/exploration-editor-page/services/user-email-preferences.service.spec.ts

```diff
@@ -0,0 +1,118 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the UserEmailPreferencesService.
+ */
+
+import { UpgradedServices } from 'services/UpgradedServices';
+
+// eslint-disable-next-line max-len
+require('pages/exploration-editor-page/services/user-email-preferences.service.ts');
+
+describe('User Email Preferences Service', function() {
+  var UserEmailPreferencesService = null;
+  var httpBackend = null;
+  var CsrfService = null;
+  var expId = '12345';
+  var sampleResponse = {
+    email_preferences: {
+      mute_feedback_notifications: false,
+      mute_suggestion_notifications: false
+    }
+  };
+
+  beforeEach(function() {
+    angular.mock.module('oppia');
+
+    angular.mock.module(function($provide) {
+      $provide.value('ExplorationDataService', {
+        explorationId: expId
+      });
+      var ugs = new UpgradedServices();
+      for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+        $provide.value(key, value);
+      }
+    });
+
+    angular.mock.inject(function($injector, $q) {
+      UserEmailPreferencesService = $injector
+        .get('UserEmailPreferencesService');
+      CsrfService = $injector.get('CsrfTokenService');
+
+      spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
+        var deferred = $q.defer();
+        deferred.resolve('sample-csrf-token');
+        return deferred.promise;
+      });
+    });
+  });
+
+  it('should successfully intialise the service', function() {
+    expect(UserEmailPreferencesService.feedbackNotificationsMuted)
+      .toBeUndefined();
+    expect(UserEmailPreferencesService.suggestionNotificationsMuted)
+      .toBeUndefined();
+
+    UserEmailPreferencesService.init(true, true);
+
+    expect(UserEmailPreferencesService.feedbackNotificationsMuted).toBe(true);
+    expect(UserEmailPreferencesService.suggestionNotificationsMuted).toBe(true);
+  });
+
+  it('should successfully return the feedbackNotificationsMuted value',
+    function() {
+      UserEmailPreferencesService.init(true, true);
+      expect(UserEmailPreferencesService.areFeedbackNotificationsMuted())
+        .toBe(true);
+    });
+
+  it('should successfully return the suggestionNotificationsMuted value',
+    function() {
+      UserEmailPreferencesService.init(true, true);
+      expect(UserEmailPreferencesService.areSuggestionNotificationsMuted())
+        .toBe(true);
+    });
+
+  describe('Testing backend related functions', function() {
+    beforeEach(angular.mock.inject(function($injector) {
```

Comment:
I think you should move all this `describe` block inside the `describe` block below. So you can inject all the services in just one `beforeEach`. Move `afterEach` and `it` to the describe block below too (and remove this one). There is not a specific reason for that, it's just to follow the way the other test files are structured.

Reply:
Hi @marianazangrossi Thanks for the review. I have implemented this, PTAL!
Thanks.